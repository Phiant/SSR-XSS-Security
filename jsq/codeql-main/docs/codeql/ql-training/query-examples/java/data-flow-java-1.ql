import java

class StringConcat extends AddExpr {
  StringConcat() { getType() instanceof TypeString }
}

from MethodCall ma
where
  ma.getMethod().getName().matches("sparql%Query") and
  ma.getArgument(0) instanceof StringConcat
select ma, "SPARQL query vulnerable to injection."
