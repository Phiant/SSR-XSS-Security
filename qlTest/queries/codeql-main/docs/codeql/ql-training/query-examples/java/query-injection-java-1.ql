import java

from Method m, MethodCall ma
where
  m.getName().matches("sparql%Query") and
  ma.getMethod() = m
select ma, m
