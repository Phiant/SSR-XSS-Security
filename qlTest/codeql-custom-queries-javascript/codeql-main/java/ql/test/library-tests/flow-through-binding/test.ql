import java
import semmle.code.java.dataflow.DataFlow

module TestConfig implements DataFlow::ConfigSig {
  predicate isSource(DataFlow::Node source) {
    source.asExpr() = any(MethodCall mc | mc.getCallee().getName() = "source")
  }

  predicate isSink(DataFlow::Node sink) {
    sink.asExpr() = any(MethodCall mc | mc.getMethod().getName() = "sink").getAnArgument()
  }
}

module Flow = DataFlow::Global<TestConfig>;

from DataFlow::Node source, DataFlow::Node sink
where Flow::flow(source, sink)
select source, sink
