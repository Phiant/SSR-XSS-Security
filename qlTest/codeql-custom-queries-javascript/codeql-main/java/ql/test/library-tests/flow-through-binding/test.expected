| Test.java:60:23:60:30 | source(...) | Test.java:61:10:61:61 | testFlowThroughSwitchStmt(...) |
| Test.java:60:23:60:30 | source(...) | Test.java:62:10:62:61 | testFlowThroughSwitchExpr(...) |
| Test.java:60:23:60:30 | source(...) | Test.java:63:10:63:68 | testFlowThroughBindingInstanceOf(...) |
| Test.java:66:51:66:58 | source(...) | Test.java:67:10:67:82 | testFlowThroughSwitchStmtWrapper(...) |
| Test.java:66:51:66:58 | source(...) | Test.java:68:10:68:82 | testFlowThroughSwitchExprWrapper(...) |
| Test.java:66:51:66:58 | source(...) | Test.java:69:10:69:89 | testFlowThroughBindingInstanceOfWrapper(...) |
